### DISCUSSION TITLE
How to create gallery of secured images

### DISCUSSION THREAD

**Discussion by pwujczyk** [igordanchenko/react-photo-album#126] [Q&A] [answered]
I would like to create a gallery that will be password protected. I have OAuth in my application and I am adding bearer to the headers to the "normal" requests. 

With the gallery I see options

- add bearer as a parameter to each picture and validate on server
- save bearer to cookie and validate on server
- generate guids for each picture and save them temporary 

Do you have any experience with those options?

**Comment by igordanchenko** [comment, marked as answer]
The solution here mostly depends on your CDN / static content backend capabilities. 

For example, if you are using AWS CloudFront you can look into implementing this with signed URLs - https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-signed-urls.html 

But if you are implementing your own backend to serve static content, you are free to chose any implementation that fits your security requirements. Some sort of pre-signed query parameter token with expiration timestamp sounds like reasonable approach to me.
